base/collection_view.go

```diff
@@ -242,8 +242,10 @@ func (c *Collection) executeViewQuery(ddoc, name string, params map[string]inter
 
 	// On timeout, return an typed error
 	if isGoCBQueryTimeoutError(err) {
+		c.releaseViewOp()
```

Comment:
Looks like we need to either releaseViewOp() on optsErr (line 238), or move the waitForAvailViewOp down to line 240.